packages/main/src/components/AnalyticalTable/index.tsx

```diff
@@ -248,6 +251,48 @@ const AnalyticalTable: FC<TableProps> = forwardRef((props: TableProps, ref: Ref<
 
   const [headerRef, tableWidth] = useWindowResize();
 
+  const [cols, setCols] = useState(columns);
```

Comment:
are you sure you need that additional state variable? This is to keep track of the current order of the columns, i suppose.
You should be able to get that through `tableState.columnOrder`.